db/db.go

```diff
@@ -4,332 +4,29 @@ import (
 	"context"
```

Comment:
You can ignore the changes here and in db_deals.go. I just refactored the code into two separate files.